query-node/schemas/council.graphql

```diff
@@ -0,0 +1,224 @@
+  intermediateWinners: [ReferendumStageRevealingOptionResult!]
+
+  "Index of current election"
+  cycleId: ElectionRound!
+}
+
+type ReferendumStageRevealingOptionResult @entity {
+  "Member that received votes."
+  optionId: Membership!
+
+  "Sum of votes' power received."
+  votePower: BigInt!
+
+  # TODO: reference variant (how?)
+  #referendumRevealingStages: [ReferendumStageRevealing!] @derivedFrom(field: "intermediateWinners")
+
+  "Election round."
+  cycleId: ElectionRound!
+
+  "Event that concluded the referendum."
+  referendumFinishedEvent: ReferendumFinishedEvent!
+}
+
+# TODO: this maybe needs to be @entity - so it's saved in db
+union ReferendumStage = ReferendumStageInactive | ReferendumStageVoting | ReferendumStageRevealing | VariantNone
+
+type CastVote @entity {
+  "Hashed vote that was casted before being revealed."
+  commitment: String!
+
+  "Election round."
+  cycleId: ElectionRound!
+
+  "Stake used to back up the vote."
+  stake: BigInt!
+
+  "Account that cast the vote."
+  castBy: String!
+
+  "Member receiving the vote."
+  voteFor: Membership
+}
+
+################### Derived ####################################################
+
+type ElectedCouncil @entity {
+  "Members that were elected to the council."
+  councilMembers: [CouncilMember!] @derivedFrom(field: "electedInCouncil")
+
+  "Changes to council status that were made during it's reign."
+  updates: [CouncilStageUpdate!] @derivedFrom(field: "electedCouncil")
+
+  "Block number at which the council was elected."
+  electedAtBlock: Int!
+
+  "Block number at which the council reign ended and a new council was elected."
+  endedAtBlock: Int
+
+  "Elections held before the council was rightfully elected."
+  councilElections: [ElectionRound!] @derivedFrom(field: "electedCouncil")
+
+  "Elections held before the next council was or will be rightfully elected."
+  nextCouncilElections: [ElectionRound!] @derivedFrom(field: "nextElectedCouncil")
+
+  # TODO: make this nullable to represent that council is already resigned https://github.com/Joystream/hydra/issues/434
+  stage: ReferendumStage!
+}
+
+type ElectionRound @entity {
+  "Election cycle ID."
+  cycleId: Int!
+
+  "Sign if election has already finished."
+  isFinished: Boolean!
+
+  "Vote cast in the election round."
+  castVotes: [CastVote!] @derivedFrom(field: "cycleId")
+
+  # TODO: reference variant (how?)
+  #referendumStageVoting: ReferendumStage @derivedFrom(field: "cycleId")
+  #referendumStageRevealing: ReferendumStage @derivedFrom(field: "cycleId")
+
+  "Council that is ruling during the election."
+  electedCouncil: ElectedCouncil!
+
+  "Council that was elected in this election round."
+  nextElectedCouncil: ElectedCouncil
+
+  "Candidates in this election round."
+  candidates: [Candidate!] @derivedFrom(field: "cycleId")
+}
+
+# TODO:
+# - is it useful as defined below?
+# - do we need to add records for BudgetRefill, BudgetIncrement, etc. or is it ok to read this info from councilEvents?
```

Comment:
In Pioneer, we're taking some of the data from the Joystream node using Polkadot API, so not from the query-node.

Reply:
Ok. In case it does makes sense, you can also query events themselves as they are saved into query node as well (as can be seen in `councilEvents.graphql`).